cardano-api/src/Cardano/Api/MetaData.hs

```diff
@@ -0,0 +1,103 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Api.MetaData
+  ( readJSONMetaData
+  , renderMetaDataError
+  ) where
+
+import           Cardano.Prelude hiding (MetaData)
+import           Prelude (String)
+
+import           Control.Exception (IOException)
+import qualified Control.Exception as Exception
+import           Control.Monad.Fail (fail)
+import           Data.Aeson
+import qualified Data.ByteString.Char8 as SC8
+import qualified Data.ByteString.Lazy.Char8 as LC8
+import qualified Data.HashMap.Strict as HMS
+import           Data.Scientific (floatingOrInteger)
+import           Data.Text.Encoding.Base64 (isBase64)
+import qualified Data.Text as Text
+import qualified Data.Vector as Vector
+
+import           Cardano.Api.Error (textShow)
+import           Shelley.Spec.Ledger.MetaData (MetaData(..), MetaDatum(..))
+
+
+data MetaDataError
+  = ConversionErrNullNotAllowed
+  | ConversionErrNumberNotInteger
+  | ConversionErrLongerThan64Bytes
+  | DecodeErrJSON !FilePath !String
+  | ReadFileErr !FilePath !String
+  deriving (Eq, Ord, Show)
+
+
+renderMetaDataError :: MetaDataError -> Text
+renderMetaDataError err =
+  case err of
+    ConversionErrNullNotAllowed -> "JSON Null value is not allowed in MetaData"
+    ConversionErrNumberNotInteger -> "Only integers are allowed in MetaData"
+    ConversionErrLongerThan64Bytes -> "JSON string is longer than 64 bytes"
+    DecodeErrJSON fp decErr ->
+      "Error decoding JSON at: " <> textShow fp <> " Error: " <> textShow decErr
+    ReadFileErr fp rErr ->
+      "Error reading file at: " <> textShow fp <> " Error: " <> textShow rErr
+
+-- No 'ToJSON' instance is written for 'MetaData'.
+-- It does not make sense to JSON roundtrip test 'MetaData'
+-- as 'String's and 'ByteString's are indistinguishable.
+instance FromJSON MetaData where
+  parseJSON = withObject "MetaData"  $ \v -> MetaData <$> (v .: "mdMap")
```

Comment:
I still think `MetaData` shouldn't be interpreted as an object containing an `mdMap` key.

That means a `MetaData` object would look like this, for example:

```json
{
  "mdMap": {
    "stuff": "this is some stuff",
    "moreStuff": 1337
  }
}
```

It seems like a `MetaData` object is really just an object in itself not requiring an `mdMap` key:

```json
{
  "stuff": "this is some stuff",
  "moreStuff": 1337
}
```